python/lsst/obs/base/ingest.py

```diff
@@ -42,28 +43,52 @@
     Formatter,
 )
 from lsst.pex.config import Config, ChoiceField, Field
-from lsst.pipe.base import Task
+from lsst.pipe.base import Task, timeMethod
 
 from ._instrument import Instrument, makeExposureRecordFromObsInfo
 from ._fitsRawFormatterBase import FitsRawFormatterBase
 
 
 def _do_nothing(*args, **kwargs) -> None:
-    """A function that accepts anything and does nothing, for use as a default
-    in callback arguments.
+    """Do nothing.
+
+    This is a function that accepts anything and does nothing.
+    For use as a default in callback arguments.
     """
     pass
 
 
+def _log_msg_counter(noun: Union[int, Iterable]) -> Tuple[int, str]:
+    """Count the iterable and return the count and plural modifier.
+
+    Parameters
+    ----------
+    noun : Iterable or `int`
+        Thing to count. If given an integer it is assumed to be the count
+        to use to calculate modifier.
+
+    Returns
```

Comment:
I'd add an `Examples` section to this, showing off your `*_log_msg_counter` trick, because I wouldn't have thought to use it like that but it's very neat.